servers/store/src/main/java/org/projectnessie/server/store/TableCommitMetaStoreWorker.java

```diff
@@ -45,7 +47,142 @@
   private final Serializer<CommitMeta> metaSerializer = new MetadataSerializer();
 
   @Override
-  public SerializerWithPayload<Contents, Contents.Type> getValueSerializer() {
+  public ByteString toStoreOnReferenceState(Contents contents) {
+    ObjectTypes.Contents.Builder builder =
+        ObjectTypes.Contents.newBuilder().setId(contents.getId());
+    if (contents instanceof IcebergTable) {
+      IcebergTable state = (IcebergTable) contents;
+      ObjectTypes.IcebergSnapshot.Builder stateBuilder =
+          ObjectTypes.IcebergSnapshot.newBuilder().setSnapshotId(state.getSnapshotId());
+      builder.setIcebergSnapshot(stateBuilder);
+
+    } else if (contents instanceof DeltaLakeTable) {
+      ObjectTypes.DeltaLakeTable.Builder table =
+          ObjectTypes.DeltaLakeTable.newBuilder()
+              .addAllMetadataLocationHistory(
+                  ((DeltaLakeTable) contents).getMetadataLocationHistory())
+              .addAllCheckpointLocationHistory(
+                  ((DeltaLakeTable) contents).getCheckpointLocationHistory());
+      String lastCheckpoint = ((DeltaLakeTable) contents).getLastCheckpoint();
+      if (lastCheckpoint != null) {
+        table.setLastCheckpoint(lastCheckpoint);
+      }
+      builder.setDeltaLakeTable(table);
+
+    } else if (contents instanceof SqlView) {
+      SqlView view = (SqlView) contents;
+      builder.setSqlView(
+          ObjectTypes.SqlView.newBuilder()
+              .setDialect(view.getDialect().name())
+              .setSqlText(view.getSqlText()));
+    } else {
+      throw new IllegalArgumentException("Unknown type " + contents);
```

Comment:
`contents.getClass().getName()`?

Reply:
it could be `null`

Reply:
true, what I mean is we should probably use the type name rather than `.toString()` in this error messge. Could we have a separate `if` for the `null` case (it looks like an error case anyway)?